File: solutions/0217_contains-duplicate/javascript/solution_presorting.js
const containsDuplicate = (nums) => {
  // presorting
  nums.sort((a, b) => a - b);

  // check if there are duplicates by comparing adjacent elements
  for (let i = 0; i < nums.length - 1; ++i) {
    if (nums[i] === nums[i + 1]) {
      return true;
    }
  }

  return false;
};

File: solutions/0217_contains-duplicate/python/solution_presorting.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # presorting
        nums.sort()

        # check if there are duplicates by comparing adjacent elements
        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                return True

        return False
